src/main/java/nl/tudelft/lifetiles/graph/models/sequence/SequenceSegment.java

```diff
@@ -154,6 +160,110 @@ public final long distanceTo(final SequenceSegment other) {
      */
     @Override
     public final int compareTo(final SequenceSegment other) {
-        return Long.compare(this.getStart(), other.getStart());
+        int candidateComp = Long.compare(this.getAbsStart(),
+                other.getAbsStart());
+        if (candidateComp == 0) {
+            candidateComp = Long.compare(this.getStart(), other.getStart());
+        }
+        if (candidateComp == 0) {
+            candidateComp = Long.compare(this.getAbsEnd(), other.getAbsEnd());
+        }
+        if (candidateComp == 0) {
+            candidateComp = Long.compare(this.getEnd(), other.getEnd());
+        }
+        if (candidateComp == 0) {
+            candidateComp = this.getContent().toString()
+                    .compareTo(other.getContent().toString());
+        }
+        if (candidateComp == 0) {
+            candidateComp = this.getSources().size()
+                    - other.getSources().size();
+        }
+        if (candidateComp == 0) {
+            Iterator<Sequence> thisIt = this.getSources().iterator();
+            Iterator<Sequence> otherIt = other.getSources().iterator();
+            while (thisIt.hasNext()) {
+                if (candidateComp == 0) {
+                    candidateComp = thisIt.next().getIdentifier()
+                            .compareTo(otherIt.next().getIdentifier());
+                }
+            }
+        }
+        return candidateComp;
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see java.lang.Object#hashCode()
+     */
+    @Override
+    public final int hashCode() {
+        final int prime = 31;
```

Comment:
Why is `prime` in the method scope and `SHIFTDISTANCE` a class constant? Also note that `SHIFTDISTANCE == prime + 1`.